at_client/lib/src/decryption_service/decryption_manager.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:at_client/src/decryption_service/decryption.dart';
+import 'package:at_client/src/decryption_service/local_key_decryption.dart';
+import 'package:at_client/src/decryption_service/self_key_decryption.dart';
+import 'package:at_client/src/decryption_service/shared_key_decryption.dart';
+import 'package:at_commons/at_commons.dart';
+
+///The manager class for [AtKeyDecryption]
+class AtKeyDecryptionManager {
+  /// Return the relevant instance of [AtKeyDecryption] for the given [AtKey]
+  static AtKeyDecryption get(AtKey atKey, String currentAtSign) {
+    // If sharedBy not equals currentAtSign, return SharedKeyDecryption
+    // Eg: currentAtSign is @bob and key is phone@alice
+    if (atKey.sharedBy != currentAtSign) {
+      return SharedKeyDecryption();
+    }
+    // Return SelfKeyDecryption for self keys.
+    // Eg: currentAtSign is @bob and _phone.wavi@bob (or) phone@bob (or) @bob:phone@bob
+    if ((atKey.sharedWith == null || atKey.sharedWith == currentAtSign) &&
+            atKey.sharedBy == currentAtSign ||
+        atKey.key!.startsWith('_')) {
```

Comment:
For readability I'd suggest either put all of `(atKey.sharedWith == null || atKey.sharedWith == currentAtSign) && atKey.sharedBy == currentAtSign` on same line of code, or wrap everything before the `||` with another set of round brackets, or both

Reply:
Added round brackets